boto3/exceptions.py

```diff
@@ -19,6 +19,32 @@ class NoVersionFound(Exception):
     pass
 
 
+class UnknownAPIVersionError(Exception):
+    def __init__(self, service_name, bad_api_version,
+                 available_api_versions):
+        msg = (
+            "The '%s' resource does not an API version of: %s\n"
```

Comment:
We might want to add this to the client in botocore as well, but that can be done in another pull request.